### DISCUSSION TITLE
"maven-enforcer-plugin (goal "enforce") is ignored by m2e" warning after upgrade to maven-enforcer-plugin 3.2.1

### DISCUSSION THREAD

**Discussion by vtintillier** [eclipse-m2e/m2e-core#1229] [Q&A]
Hello,

I am not sure if the behavior I see is intended or not.

`maven-enforcer-plugin` removed their `META-INF/m2e/lifecycle-mapping-metadata.xml` file in their last 3.2.1 release (https://github.com/apache/maven-enforcer/pull/192). As a result, we see this warning in all our poms:

```
maven-enforcer-plugin (goal "enforce") is ignored by m2e
```

![image](https://user-images.githubusercontent.com/494652/216346051-227be72d-9152-4b96-831c-c230f8598d12.png)

It seems this messages originate from this code:

https://github.com/eclipse-m2e/m2e-core/blob/e3794059b1250729872daecf9c1ff0b35fc75daf/org.eclipse.m2e.core/lifecycle-mapping-metadata.xml#L120-L134

So my question is: is this expected that goals ignored in `m2e-core` own `lifecycle-mapping-metadata.xml` appear as warnings in the editor, whereas goals ignored in maven plugins own `lifecycle-mapping-metadata.xml` do not?
Also is there a way to disable those warnings?

Thanks
Vivien

**Comment by kwin** [comment]
I think the `message` part inside the `ignore` lets m2e emit that warning. I don't think this warning is useful but rather confusing therefore I propose to remove all message lines from https://github.com/eclipse-m2e/m2e-core/blob/e3794059b1250729872daecf9c1ff0b35fc75daf/org.eclipse.m2e.core/lifecycle-mapping-metadata.xml#L131. @laeubi WDYT?

**Reply by laeubi** [reply]
I think the only useful thing would be if m2e suggest something actionable, e.g. 

> m2 does not support maven-enforcer-plugin (goal "enforce") with version X please upgrade to at least version Y to get full support

But if we choose that ignore is ok and the user can't do anything anyways , the message can be removed.

**Reply by kwin** [reply]
Alternatively we can lower the severity for those messages just to make the user aware of the status quo, but this isn't anything to fix (i.e. ERROR or WARN is just wrong).

**Reply by laeubi** [reply]
Yep, but the question is, as a user, what should I do with such a message? e.g. are there cases where I do NOT want to ignore that goal?

**Reply by vtintillier** [reply]
@kwin  @laeubi So is it OK if I create an issue from this discussion, and create a PR where I remove the messages from the `<ignore>` elements?

**Reply by laeubi** [reply]
Sure.
